### ISSUE TITLE
Have teams for each "skill"

### ISSUE THREAD

**Issue by untitaker** [jazzband/help#10]
I currently get pinged via "jazzband/members" to review Django projects. While I would love to help, I have never used Django at all.

Proposal: Create teams for each "skill". There'd be a team for "Django", one for "Python", one for "Security"... Then everybody can join/subscribe to the teams they feel competent enough for, and people can CC those teams for more targeted calls for help.

**Comment by jezdez** [comment]
Interesting! I like that idea.

It would assume that it'd require a join/leave UI on the Jazzband site since I can't offer org admin permissions to join/leave teams for all members for security reasons.

But since the site already has effectively implemented GitHub auth, it's just a matter of the right API calls to do it.

**Comment by jezdez** [comment]
To be clear you're not proposing to actually assign different repos to each team and by that limiting the access permissions again for members "outside of their skillset"?

**Comment by untitaker** [comment]
>To be clear you're not proposing to actually assign different repos to each team and by that limiting the access permissions again for members "outside of their skillset"?

No, didn't think of that, though I guess it's a problem sometime in the future. Perhaps both problems could be solved at once, not sure.

**Comment by jezdez** [comment]
@untitaker Hey there again, took me a while but I've been thinking about this more recently since I want to improve the unassisted releases to PyPI which in turn requires improvements to the site code. While working on this I found a way to better assign members to projects and wondered if you'd still consider "skill teams" (or "genre" when staying in the musical analogy) worthwhile?

**Comment by untitaker** [comment]
I'm unsure whether picking vague metaphors is a good idea from a UX perspective.

Do you propose something based on Github teams, or do you want to let users manage that through the site's UI?

**Comment by jezdez** [comment]
So in https://github.com/jazzband/website/commit/8bac98e1a824989d2c464ccd91fdf6ba2290555d I've added per-project memberships, which I will map to GitHub teams so it's easier to be notified per-project with @jazzband/my-project etc. The UI hasn't been updated to show the membership feature, since I want to populate it with the contributors of each individual project (git commits, issues etc) and I haven't written that code yet.

But, I wonder if you're still interested in general of having per-topics-teams or if you have changed your mind since then?

**Comment by untitaker** [comment]
Unfortunately I don't really use Jazzband anymore, but at the time it seemed
like a good idea to have project-overarching interest groups (particularly for
"security" topic)

**Comment by jezdez** [comment]
@untitaker Ah, bummer, thanks for the idea though, together with the [per-team-discussion boards](https://github.com/blog/2471-introducing-team-discussions) this could be a neat thing to do.

**Comment by bittner** [comment]
I feel the topic around skills makes sense. For example, if we would assign, say, CI-related PRs to a CI/CD skills team for review everyone in there would get a notification.

We can still have separate per-team-discussion boards for the very same skills teams, but I see the main value of skills teams in subscribing to tailored notifications.

**Comment by tim-schilling** [comment]
I for one would really like being able to ask for a review for CI and CD are setup. My usual task is, "get it working", then ignore it until it becomes a problem again. Having the ability to request a review from people who have more expertise would help folks like myself learn some of those best practices via these reviews.

To that end, would creating Github teams be sufficient? Is this something that should be exposed outside of Jazzband? If so, then would that mean we'd want to make them a project team on the jazzband site?

Personally, I'd prefer to keep it simpler with the github teams only implementation to start with. However, I suspect there are other considerations I'm ignorant of. @jezdez what can you foresee as issues with a github teams only implementation of skill based teams?

**Comment by webknjaz** [comment]
I think this might work to an extent. Also, there's a CI/CD role with certain administrative privileges if there's a need to have a group with elevated trust: https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/using-organization-roles.
